libs/src/services/solanaWeb3Manager/index.js

```diff
@@ -0,0 +1,168 @@
+const solanaWeb3 = require('@solana/web3.js')
+const { transferWAudioBalance } = require('./transfer')
+const { getBankAccountAddress, createUserBankFrom, getBankAccountInfo } = require('./userBank')
+const { wAudioFromWeiAudio } = require('./wAudio')
+
+const { PublicKey } = solanaWeb3
+
+/**
+ * SolanaWeb3Manager acts as the interface to solana contracts from a client.
+ * It wraps methods to create and lookup user banks, transfer balances, and
+ * interact with the @solana/web3 library.
+ *
+ * Note: Callers of this class should specify all $AUDIO amounts in units of wei.
+ * The internals of this class should handle the conversion from wei AUDIO to wormhole
+ * $AUDIO amounts.
+ */
+class SolanaWeb3Manager {
+  /**
+   * @param {Object} solanaWeb3Config
+   * @param {string} solanaWeb3Config.solanaClusterEndpoint
+   *  the solana cluster RPC endpoint
+   * @param {string} solanaWeb3Config.mintAddress
+   *  address for the wAudio mint - Audius (Wormhole) token
+   * @param {string} solanaWeb3Config.solanaTokenAddress
+   *  the SPL token program address
+   * @param {string} solanaWeb3Config.claimableTokenPDA
+   *  the program derived address for the claimable token program (using the mint)
+   * @param {string} solanaWeb3Config.feePayerAddress
+   *  the address for the specific fee payer to be used in relayed transactions
+   * @param {string} solanaWeb3Config.claimableTokenProgramAddress
+   *  the address for the claimable token program used to create banks and transfer wAudio
+   *
+   * @param {IdentityService} identityService
+   * @param {Web3Manager} web3Manager
+   */
+  constructor (
+    solanaWeb3Config,
+    identityService,
+    web3Manager
+  ) {
+    this.solanaWeb3Config = solanaWeb3Config
+    this.identityService = identityService
+    this.web3Manager = web3Manager
+
+    this.web3 = solanaWeb3
```

Comment:
Maybe a little confusing that `web3Manger` is for eth and `web3` is the solanaWeb3, but probs fine.

Reply:
let's do solanaWeb3 good call